tests/test_auth.py

```diff
@@ -0,0 +1,77 @@
+import json
+import unittest
+
+from app import db
+from app.models import User
+from tests.test_resources import BaseTest
+
+
+class TestAuthorization(BaseTest):
+
+    def test_register_succesfully(self):
+        endpoint = '/api/v1/auth/register'
+        data = {'username': 'dee', 'password': 'hello'}
+
+        response = self.client.post(endpoint, data=data)
+        welcome_message = json.loads(response.get_data(as_text=True)).get('message')
+
+        self.assertEqual("Welcome dee to the Bucketlist Service.", welcome_message)
+        self.assertEqual(response.status_code, 200)
+
+    def test_register_user_persistence(self):
+        endpoint = '/api/v1/auth/register'
+        data = {'username': 'dee', 'password': 'hello'}
+        user = User.query.filter_by(username='dee').first()
+        self.assertIsNone(user)
+
+        self.client.post(endpoint, data=data)
+        user = User.query.filter_by(username='dee').first()
+
+        self.assertIsNotNone(user)
+        self.assertIn('dee', user.username)
+
+    def test_register_with_exisitng_username(self):
+        db.session.add(self.user)
+        db.session.commit()
+        endpoint = '/api/v1/auth/register'
+        data = {'username': 'damidee', 'password': 'andela'}
```

Comment:
maybe have this username as `self.user.username` so it is clear we are creating with an existing username.